### PR TITLE
Apps: Add clean scripts

### PR DESCRIPTION
**Pull request by sirreal**
Adds `clean` scripts following conventions from other packages and apps in the monorepo.

#### Changes proposed in this Pull Request

* All the apps now have clean scripts and follow a convention.

#### Testing instructions

* `npx lerna run clean` should result in apps being cleaned up
* Clean scripts should run before relevant scripts, like build and dev.
* Artifacts produced by relevant CircleCI jobs should be unchanged.

Noticed in #36583

### CODE REVIEW

**Review by simison — APPROVED** [review]
Looks good and CI passes.

"Owner" teams could still give thumbsup before merging tho.

File: apps/full-site-editing/package.json [hunk 1]
```diff
@@ -25,11 +25,14 @@
 		"starter-page-templates": "calypso-build --source='starter-page-templates'",
 		"dev:starter-page-templates": "npm run starter-page-templates",
 		"build:starter-page-templates": "NODE_ENV=production npm run starter-page-templates",
-		"dev": "rimraf dist && npm-run-all --parallel dev:*",
-		"build": "rimraf dist && npm-run-all --parallel build:*",
+		"dev": "npm-run-all --parallel dev:*",
+		"build": "npm-run-all --parallel build:*",
```
Comment (simison) [inline apps/full-site-editing/package.json]: I really appreciate this long form instead of `run-p` 👍  ...should do the same in the root `package.json` :-)

**Review by vindl — APPROVED** [review]
Thanks for adding this! LGTM! :shipit:

One thing that I've noticed is that Artifacts tab (where you could browse the generated folder structure) is no longer present for these jobs, but that seems to be a recent change on the CircleCI side? In any case I verified that our sync scripts can still fetch them when passed the the appropriate job ID.

**Comment by mmtr** [comment]
> One thing that I've noticed is that Artifacts tab (where you could browse the generated folder structure) is no longer present for these jobs, but that seems to be a recent change on the CircleCI side? In any case I verified that our sync scripts can still fetch them when passed the the appropriate job ID.

That happened to me in the past when I was logged out of CircleCI. Make sure you're authenticated with GitHub (https://circleci.com/vcs-authorize/) and you should see the Artifacts tab again.

**Comment by vindl** [comment]
> Make sure you're authenticated with GitHub (https://circleci.com/vcs-authorize/) and you should see the Artifacts tab again.

Yup, that was it. Thanks!

One more note regarding the PR - this doesn't seem to be deleting `dist` folders in all of the appropriate FSE places. Due to placing multiple projects into a single plugin, every one of them has a separate `dist` folder (Full Site Editing, Starter Page Templates, Posts List Block). These were supposed to be split out into separate plugins, but the work on that has been punted recently.

File: apps/full-site-editing/package.json [hunk 2]
```diff
@@ -25,11 +25,14 @@
...
 		"test:unit": "wp-scripts test-unit-js --config='jest.config.js'",
 		"test:unit:help": "wp-scripts test-unit-js --config='jest.config.js' --help",
-		"test:unit:watch": "wp-scripts test-unit-js --config='jest.config.js' --watch"
+		"test:unit:watch": "wp-scripts test-unit-js --config='jest.config.js' --watch",
+		"clean": "npx rimraf dist",
```
Comment (sirreal) [inline apps/full-site-editing/package.json]: It sounds like this clean is incomplete and a couple more directories should be removed.

Via @vindl https://github.com/Automattic/wp-calypso/pull/36602#issuecomment-539942350

Comment (sirreal) [reply apps/full-site-editing/package.json]: I tracked these down and added them to clean in 6376c95f839f39c8f399ec07ec1b774ad2d2c2ca

**Comment by sirreal** [comment]
Rebased.

I added additional directories to clean and adjusted the way we handle ignores (standard `dist` directory ignored globallay, custom `dist` directories ignored in the app).

This should be ready to merge. If anyone has feedback, let me know. Otherwise I'll plan to merge early next week.

**Pull request merged**